### DISCUSSION TITLE
Text Data Structure

### DISCUSSION THREAD

**Discussion by nerocui** [WelterDevelopment/CodeWriter-WinUI#9] [Q&A]
Is this control using rope data structure? There only one file with 3000+ lines of code, so it's a little hard to figure out the structure.

**Comment by WelterDevelopment** [comment]
I know, the code is a mess... sorry!
I am not a professional programmer. The whole text document (with all calculated tokens) is stored in 
``` csharp
public ObservableCollection<Line> Lines { get => Get(new ObservableCollection<Line>()); set => Set(value); }
```
With ```Line.SetLineText(string value)``` the Regex-based tokenization just recalculates the whole line. I have no idea how to implement rope data structure and how to do the tokenization then.

In ~1 month I have hollidays, then I will do code refactoring and I will work on some more intelligent ways of tokenization and data handling. I added the rope idea to my list ;)
